uaclient/api/u/pro/attach/auto/full_auto_attach/v1.py

```diff
@@ -69,24 +69,29 @@ def full_auto_attach(options: FullAutoAttachOptions) -> FullAutoAttachResult:
 
 
 def _full_auto_attach(
-    options: FullAutoAttachOptions, cfg: UAConfig
+    options: FullAutoAttachOptions,
+    cfg: UAConfig,
+    *,
+    mode: event_logger.EventLoggerMode = event_logger.EventLoggerMode.JSON
 ) -> FullAutoAttachResult:
     try:
         with lock.SpinLock(
             cfg=cfg,
             lock_holder="pro.api.u.pro.attach.auto.full_auto_attach.v1",
         ):
-            ret = _full_auto_attach_in_lock(options, cfg)
+            ret = _full_auto_attach_in_lock(options, cfg, mode=mode)
     except Exception as e:
         lock.clear_lock_file_if_present()
         raise e
     return ret
 
 
 def _full_auto_attach_in_lock(
-    options: FullAutoAttachOptions, cfg: UAConfig
+    options: FullAutoAttachOptions,
+    cfg: UAConfig,
+    mode: event_logger.EventLoggerMode,
 ) -> FullAutoAttachResult:
-    event.set_event_mode(event_logger.EventLoggerMode.JSON)
+    event.set_event_mode(mode)
```

Comment:
I know why you are setting the `event_logger` here, but I wonder if we actually need to set this module here. Do we need this setup anymore @orndorffgrant @renanrodrigo @CalvoM ? It seems we are not even using it on the `actions` module anymore

Reply:
I think `event.info` is used during `enable` which is why we still need this.

So I think we'll need to migrate enable to the api structure before we can remove this

Reply:
Yes, the reason to add this as a configurable parameter is to be able to set the `event_logger` in CLI mode when we use it from `uaclient.cli` and print out events to stdout and stderr, as we do without this refactoring.